### PR TITLE
1709 rate cost type

### PR DESCRIPTION
**Pull request by adberglund**
## Summary
- Allow a cost type on rates for cost model price lists. Acceptable values are `Infrastructure` or `Supplementary`.
- Added default cost types to the cost model metrics API. 

## Testing

```
GET /api/cost-management/v1/costmodels/
HTTP 200 OK
Allow: GET, POST, HEAD, OPTIONS
Cache-Control: max-age=0, no-cache, no-store, must-revalidate
Content-Type: application/json
Expires: Wed, 26 Feb 2020 20:12:10 GMT
Vary: Accept

{
    "meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/costmodels/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/costmodels/?limit=10&offset=0"
    },
    "data": [
        {
            "uuid": "c0383cf6-d412-46a0-8459-7fd2ff3a7201",
            "name": "Cost Management OpenShift Cost Model",
            "description": "A cost model of on-premises OpenShift clusters.",
            "source_type": "OpenShift Container Platform",
            "created_timestamp": "2020-02-26T20:11:34.123352Z",
            "updated_timestamp": "2020-02-26T20:11:34.123392Z",
            "rates": [
                {
                    "metric": {
                        "name": "node_cost_per_month",
                        "label_metric": "Node",
                        "label_measurement": "Currency",
                        "label_measurement_unit": "node-month"
                    },
                    "tiered_rates": [
                        {
                            "unit": "USD",
                            "value": 1000.0,
                            "cost_type": "Infrastructure",
                            "usage": {
                                "usage_start": null,
                                "usage_end": null,
                                "unit": "USD"
                            }
                        }
                    ]
                }
            ],
            "markup": {},
            "providers": []
        }
    ]
}

POST /api/cost-management/v1/costmodels/
HTTP 201 Created
Allow: GET, POST, HEAD, OPTIONS
Cache-Control: max-age=0, no-cache, no-store, must-revalidate
Content-Type: application/json
Expires: Wed, 26 Feb 2020 20:12:41 GMT
Vary: Accept

{
    "uuid": "14ef75a4-63c9-4ec2-9777-7af547039ff0",
    "name": "Cost Management OpenShift Cost Model",
    "description": "A cost model of on-premises OpenShift clusters.",
    "source_type": "OpenShift Container Platform",
    "created_timestamp": "2020-02-26T20:12:41.802292Z",
    "updated_timestamp": "2020-02-26T20:12:41.802332Z",
    "rates": [
        {
            "metric": {
                "name": "node_cost_per_month",
                "label_metric": "Node",
                "label_measurement": "Currency",
                "label_measurement_unit": "node-month"
            },
            "tiered_rates": [
                {
                    "value": 1000.0,
                    "unit": "USD",
                    "cost_type": "Supplementary",
                    "usage": {
                        "usage_start": null,
                        "usage_end": null,
                        "unit": "USD"
                    }
                }
            ]
        }
    ],
    "markup": {},
    "providers": []
}

HTTP 400 Bad Request
Allow: GET, POST, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "errors": [
        {
            "detail": "\"Infrastructurezzz\" is not a valid choice.",
            "source": "rates.cost_type",
            "status": 400
        }
    ]
}
```

### CODE REVIEW

File: koku/api/migrations/0009_providerstatus_squashed_0042_auto_20200116_2048.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def load_openshift_metric_map(apps, schema_editor):
     data = json.loads(data)
 
     for entry in data:
+        del entry["default_cost_type"]
```
Comment (Red-HAP) [inline koku/api/migrations/0009_providerstatus_squashed_0042_auto_20200116_2048.py]: Please add a comment here about why the data is being deleted during this migration. This is something that could be easily missed if migrations are super-squashed as they were last time.

File: koku/api/metrics/models.py [hunk 2]
```diff
@@ -56,3 +56,5 @@ class Meta:
     label_measurement = models.CharField(max_length=256, null=False)
 
     label_measurement_unit = models.CharField(max_length=64, null=False)
+
+    default_cost_type = models.TextField(null=True)
```
Comment (Red-HAP) [inline koku/api/metrics/models.py]: Is the new strategy to **not** add the choices at the model definition?
Also: should a default be set here to match the default used below in the serializer?

File: koku/api/migrations/0009_providerstatus_squashed_0042_auto_20200116_2048.py (hunk 1 above)
Comment (adberglund) [inline koku/api/migrations/0009_providerstatus_squashed_0042_auto_20200116_2048.py]:
```suggestion
        # Deleting this entry as it does not exist in the table/model at this point in migrations
        del entry["default_cost_type"]
```

Comment (adberglund) [reply koku/api/migrations/0009_providerstatus_squashed_0042_auto_20200116_2048.py]: Got it

**Pull request merged**